src/script/chem/molfile/molfile.js

```diff
@@ -286,9 +286,135 @@ Molfile.prototype.writeCTab3000 = function (molecule) { // eslint-disable-line m
 	this.writeCR(`M  V30 COUNTS ${noOfAtoms} ${noOfBonds} ${noOfSGroups} 0 ${isChiral}`);
 	this.writeAtomBlock3000(molecule);
 	this.writeBondBlock3000(molecule);
+	this.writeSGroupBlock3000(molecule);
 	this.writeCR('M  V30 END CTAB');
 };
 
+Molfile.prototype.writeSGroupBlock3000 = function (molecule) {
+	if (molecule.sgroups.size === 0) {
+		return;
+	}
+	this.writeCR('M  V30 BEGIN SGROUP');
+	var sgmap = {};
+	var cnt = 1;
+	var sgmapback = {};
+	var sgorder = molecule.sGroupForest.getSGroupsBFS();
+	sgorder.forEach((id) => {
+		sgmapback[cnt] = id;
+		sgmap[id] = cnt++;
```

Comment:
I'm not sure how others feel here, but I'd prefer to have ++increments inside a for loop, or use the second {{index}} arg of the forEach to keep track of index.

Reply:
Done